packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -731,6 +755,27 @@ class ScrollableState extends State<Scrollable> with TickerProviderStateMixin, R
   String? get restorationId => widget.restorationId;
 }
 
+/// Describes the aspects of a Scrollable widget to inform inherited widgets
+/// like [ScrollBehavior] for decorating.
+///
+/// Decorations like [GlowingOverscrollIndicator]s and [Scrollbar]s require
+/// information about the Scrollable in order to be initialized.
+@immutable
+class ScrollableDetails {
+  /// Creates a set of details describing the [Scrollable]. The [direction]
+  /// cannot be null.
+  const ScrollableDetails(this.direction, this.controller);
+
+  /// The direction in which this widget scrolls.
+  ///
+  /// Cannot be null.
+  final AxisDirection direction;
+
+  /// A [ScrollController] that can be used to control the position of the
+  /// [Scrollable] widget.
```

Comment:
Document what it means to set this to null or non-null?

Reply:
Maybe we should even rename this to make it clear that this is the controller for the scrollbar?

Reply:
I am hesitant to name this as specific to scrollbars, the controller may be used in the future for other things.